### PR TITLE
CS1/CS2 - Merge and improve disputes module

### PR DESCRIPTION
**Pull request by Armando-Ibarra**

### CODE REVIEW

File: src/main/java/com/checkout/disputes/DisputesClient.java [hunk 1]
```diff
@@ -10,15 +10,15 @@
...
-    CompletableFuture<DisputeDetailsResponse> getDisputeDetails(String id);
+    CompletableFuture<DisputeDetailsResponse> getDisputeDetails(String disputeId);
 
-    CompletableFuture<Void> accept(String id);
+    CompletableFuture<Void> accept(String disputeId);
 
     CompletableFuture<Void> putEvidence(String id, DisputeEvidenceRequest disputeEvidence);
 
-    CompletableFuture<DisputeEvidenceResponse> getEvidence(String id);
+    CompletableFuture<DisputeEvidenceResponse> getEvidence(String disputeId);
```
Comment (rsaestrela) [inline src/main/java/com/checkout/disputes/DisputesClient.java:19]: Should the parameter be `disputeId`? Same below.

File: src/main/java/com/checkout/disputes/four/DisputesClient.java [hunk 2]
```diff
@@ -3,18 +3,20 @@
...
+import com.checkout.disputes.DisputeEvidenceResponse;
 
 import java.util.concurrent.CompletableFuture;
 
 public interface DisputesClient {
 
     CompletableFuture<DisputesQueryResponse> query(DisputesQueryFilter queryFilter);
 
-    CompletableFuture<DisputeDetailsResponse> getDisputeDetails(String disputeId);
+    CompletableFuture<DisputeDetailsResponse> disputeDetails(String disputeId);
```
Comment (rsaestrela) [inline src/main/java/com/checkout/disputes/four/DisputesClient.java:15]: Why the different name? It's not possible to override the method from the parent?

File: src/main/java/com/checkout/disputes/four/DisputesClientImpl.java [hunk 3]
```diff
@@ -1,21 +1,14 @@
...
 import com.checkout.CheckoutConfiguration;
 import com.checkout.SdkAuthorizationType;
-import com.checkout.common.FileDetailsResponse;
-import com.checkout.common.FileRequest;
-import com.checkout.common.IdResponse;
 
 import java.util.concurrent.CompletableFuture;
 
-public class DisputesClientImpl extends AbstractClient implements DisputesClient {
+public class DisputesClientImpl extends com.checkout.disputes.DisputesClientImpl implements DisputesClient {
```
Comment (rsaestrela) [inline src/main/java/com/checkout/disputes/four/DisputesClientImpl.java:9]: Don't really like this approach, but I think it's a great improvement and a step forward in terms of code reuse on this module. 🤙

File: src/main/java/com/checkout/disputes/DisputesClientImpl.java [hunk 4]
```diff
@@ -12,15 +12,19 @@
...
+    private static final String DISPUTES = "disputes";
+    private static final String FILES = "files";
     private static final String ACCEPT = "accept";
     private static final String EVIDENCE = "evidence";
 
     public DisputesClientImpl(final ApiClient apiClient, final CheckoutConfiguration configuration) {
         super(apiClient, configuration, SdkAuthorizationType.SECRET_KEY);
     }
 
+    protected DisputesClientImpl(final ApiClient apiClient, final CheckoutConfiguration configuration, final SdkAuthorizationType sdkAuthorizationType) {
```
Comment (rsaestrela) [inline src/main/java/com/checkout/disputes/DisputesClientImpl.java:24]: So, basically the extra constructor is to set a different `SdkAuthorizationType`?

**Review by rsaestrela — CHANGES_REQUESTED** [review]
Nice work. Please rebase from master, you should get changes in testing that will help to get a green build here.

File: src/main/java/com/checkout/disputes/DisputesQueryFilter.java [hunk 5]
```diff
@@ -1,26 +1,62 @@
...
+
+    @SerializedName("payment_reference")
     private String paymentReference;
+
+    @SerializedName("payment_arn")
     private String paymentArn;
-    private String thisChannelOnly;
+
+    @SerializedName("this_channel_only")
+    private boolean thisChannelOnly;
```
Comment (rsaestrela) [inline src/main/java/com/checkout/disputes/DisputesQueryFilter.java:41]: Should this be a `Boolean`?

File: src/main/java/com/checkout/disputes/DisputesQueryFilter.java [hunk 6]
```diff
@@ -1,26 +1,62 @@
...
+
+    /**
+     * Available only on Four
+     */
+
+    /**
+     * One or more comma-separated client entities. This works like a logical OR operator
+     */
+    @SerializedName("entity_ids")
+    private String entityIds;
```
Comment (rsaestrela) [inline src/main/java/com/checkout/disputes/DisputesQueryFilter.java:51]: Can we make this a `List` and then join them with "," so users don't have to do it? Same for `subEntityIds`.

**Review by rsaestrela — APPROVED** [review]
Some minor improvements identified. Nice work!

File: src/main/java/com/checkout/disputes/DisputesQueryFilter.java (hunk 5 above)
Comment (Armando-Ibarra) [reply src/main/java/com/checkout/disputes/DisputesQueryFilter.java:41]: Yes, it is documented [here](https://api-reference.checkout.com/preview/crusoe/#tag/Disputes/paths/~1disputes/get)

**Pull request merged**